Build scripts/instructions: trouble with tkchia ppa

**Describe the bug**

Preparing to build from source, early setup adds the `tkchia` PPA.  After enabling it, `apt update` fails.

Inspection of the ppa, specifically http://ppa.launchpad.net/tkchia/build-ia16/ubuntu/dists/ shows no entry for kinetic, which is what the sources.list entry seeks.

**To Reproduce**

0. This is on a Debian 11 (stable, bullseye) system, amd64 architecture
1. As previously [reported](https://github.com/dosemu2/dosemu2/issues/1649), `add-apt-repository` doesn't work, at least with Ubuntu ppa's on this system.  Since the setup script executes this command as the first step, that's a problem.
2. Fix the [bug in add-apt-repository](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1012649).  See the report there for a much fuller analysis and suggested fix.  Eventually I'll get the patch I'm using up there (also available below).
3. Having manually taken care of the first repository, `dosemu2`, while diagnosing the problem above, complete the steps at the start of `ci_prereq.sh`:  
```shell
sudo add-apt-repository -y ppa:jwt27/djgpp-toolchain
sudo add-apt-repository -y ppa:tkchia/build-ia16
```
These run without complaints.
4. The next step in the script is to update the package information:
```shell
# date; apt update
Tue 14 Jun 2022 10:53:39 PM PDT
Hit:1 http://localhost:3142/debian bullseye InRelease
Hit:2 https://packages.microsoft.com/repos/vscode stable InRelease
Hit:3 http://localhost:3142/debian-security bullseye-security InRelease
Hit:4 http://ppa.launchpad.net/dosemu2/ppa/ubuntu kinetic InRelease
Get:5 http://ppa.launchpad.net/jwt27/djgpp-toolchain/ubuntu kinetic InRelease [18.1 kB]
Ign:6 http://ppa.launchpad.net/tkchia/build-ia16/ubuntu kinetic InRelease
Get:7 http://ppa.launchpad.net/jwt27/djgpp-toolchain/ubuntu kinetic/main i386 Packages [1,040 B]
Get:8 http://ppa.launchpad.net/jwt27/djgpp-toolchain/ubuntu kinetic/main amd64 Packages [1,712 B]
Get:9 http://ppa.launchpad.net/jwt27/djgpp-toolchain/ubuntu kinetic/main Translation-en [652 B]
Err:10 http://ppa.launchpad.net/tkchia/build-ia16/ubuntu kinetic Release
  404  Not Found [IP: 185.125.190.52 80]
Reading package lists... Done
E: The repository 'http://ppa.launchpad.net/tkchia/build-ia16/ubuntu kinetic Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```
5. Verify the `tkchia` distributions do not have a `kinetic` entry

**Comments**

1. The instructions and the commands to add the repositories do not mention `kinetic`.  It seems  likely that `add-apt-repository` came up with it on its own as either the current release, or the release that was current when Debian 11 was frozen (pretty recently).
2. So one solution might be to change the release name in the sources.list entry.  What it should be, and whether they all need to be the same, I don't know.  My guess is all 3 PPA's should be using the same release.  This involves manually editing after the fact.
3. A longer-term fix would be to change the script and instructions to specify an appropriate release.  I *suspect* `add-apt-repository` may support this because the man page says "In the third form, the given distribution component will be enabled for all sources."  But the man page is badly mangled--there is no 3rd form shown--and I'm not sure if "component" and "release" are the same things.
4. The error message suggests that relaxing security might be a solution.  I don't think that would help at all in this case, since the file, and the accompanying release, is just absent.
5. Another "solution" would be for `tkchia` to add the `kinetic` release.

**The Patch**
`add-apt-repository` is from the `software-properties` source package.  It ends up calling `ppa.py`, which is what needs the fix.  Starting at line 270, my modified version is
```python
        res = subprocess.call(["apt-key", "--keyring", apt_keyring, "add",
            tmp_export_keyring])
```
In the original, the last argument is `tmp_keyring`, which fails because that file is in keybox format.
Newer versions of `gpg` exhibit this pickiness; I gather older ones didn't.

I add my thanks to @skitt: I had tried `apt install` with a deb and failed; I didn't know I needed a `/` to get the correct interpretation.

The instructions in https://github.com/dosemu2/dosemu2/wiki/Building-packages-for-Debian may now be getting me to the expected compiler error.  That means I can't actually build the package (still). Would using `clang` work better?
I get to the step of building dosemu2 and see

```
gcc -c -imacros config.hh -MD -DCFLAGS_STR=" -fplan9-extensions -Wall -Wstrict-prototypes -Wmissing-declarations -Wnested-externs -fms-extensions -pthread -Wno-unused-result -Wcast-qual -Wwrite-strings -Wstrict-aliasing=2 -Wno-address-of-packed-member -ggdb3 -fpie   -g -O2 -ffile-prefix-map=/usr/local/src/dos/dosemu2=. -fstack-protector-strong -Wformat -Werror=format-security" -I../../../src/include -I../../../src/plugin/include -I../../../src/base/bios/x86 -I/usr/local/src/dos/dosemu2/build/../src/include -I/usr/local/src/dos/dosemu2/build/../src/base/lib -Wdate-time -D_FORTIFY_SOURCE=2 -fplan9-extensions -Wall -Wstrict-prototypes -Wmissing-declarations -Wnested-externs -fms-extensions -pthread -Wno-unused-result -Wcast-qual -Wwrite-strings -Wstrict-aliasing=2 -Wno-address-of-packed-member -ggdb3 -fpie   -g -O2 -ffile-prefix-map=/usr/local/src/dos/dosemu2=. -fstack-protector-strong -Wformat -Werror=format-security -o render.o /usr/local/src/dos/dosemu2/build/../src/base/video/render.c
during RTL pass: sched2
/usr/local/src/dos/dosemu2/build/../src/base/core/int.c: In function ‘int33_unrevect_fixup’:
/usr/local/src/dos/dosemu2/build/../src/base/core/int.c:1748:1: internal compiler error: in move_insn, at haifa-sched.c:5471
 1748 | }
      | ^

## leading to
make[2]: Leaving directory '/usr/local/src/dos/dosemu2/build/src'
make[1]: *** [Makefile:30: default] Error 2
make[1]: Leaving directory '/usr/local/src/dos/dosemu2/build'
dh_auto_build: error: cd build && make -j8 returned exit code 2
make: *** [debian/rules:6: build] Error 25
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
debuild: fatal error at line 1182:
dpkg-buildpackage -us -uc -ui -i -b failed
make: *** [Makefile:65: deb] Error 29

```

There are also some warnings like
```
gcc -c -imacros config.hh -MD -DCFLAGS_STR=" -fplan9-extensions -Wall -Wstrict-prototypes -Wmissing-declarations -Wnested-externs -fms-extensions -pthread -Wno-unused-result -Wcast-qual -Wwrite-strings -Wstrict-aliasing=2 -Wno-address-of-packed-member -ggdb3 -fpie   -g -O2 -ffile-prefix-map=/usr/local/src/dos/dosemu2=. -fstack-protector-strong -Wformat -Werror=format-security" -I../../../../src/include -I../../../../src/plugin/include -I../../../../src/base/bios/x86 -I/usr/local/src/dos/dosemu2/build/../src/include -I/usr/local/src/dos/dosemu2/build/../src/base/lib -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/local/src/dos/dosemu2/build/../src/base/lib/mcontext -fplan9-extensions -Wall -Wstrict-prototypes -Wmissing-declarations -Wnested-externs -fms-extensions -pthread -Wno-unused-result -Wcast-qual -Wwrite-strings -Wstrict-aliasing=2 -Wno-address-of-packed-member -ggdb3 -fpie   -g -O2 -ffile-prefix-map=/usr/local/src/dos/dosemu2=. -fstack-protector-strong -Wformat -Werror=format-security -o debug.o /usr/local/src/dos/dosemu2/build/../src/arch/linux/async/debug.c
gcc -c -imacros config.hh -MD -DCFLAGS_STR=" -fplan9-extensions -Wall -Wstrict-prototypes -Wmissing-declarations -Wnested-externs -fms-extensions -pthread -Wno-unused-result -Wcast-qual -Wwrite-strings -Wstrict-aliasing=2 -Wno-address-of-packed-member -ggdb3 -fpie   -g -O2 -ffile-prefix-map=/usr/local/src/dos/dosemu2=. -fstack-protector-strong -Wformat -Werror=format-security" -I../../../src/include -I../../../src/plugin/include -I../../../src/base/bios/x86 -I/usr/local/src/dos/dosemu2/build/../src/include -I/usr/local/src/dos/dosemu2/build/../src/base/lib -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/local/src/dos/dosemu2/build/../src/include/keyboard -fplan9-extensions -Wall -Wstrict-prototypes -Wmissing-declarations -Wnested-externs -fms-extensions -pthread -Wno-unused-result -Wcast-qual -Wwrite-strings -Wstrict-aliasing=2 -Wno-address-of-packed-member -ggdb3 -fpie   -g -O2 -ffile-prefix-map=/usr/local/src/dos/dosemu2=. -fstack-protector-strong -Wformat -Werror=format-security -o keymaps.o /usr/local/src/dos/dosemu2/build/../src/base/kbd_unicode/keymaps.c
/usr/local/src/dos/dosemu2/build/../src/base/lib/misc/dlmalloc.c: In function ‘init_bins’:
/usr/local/src/dos/dosemu2/build/../src/base/lib/misc/dlmalloc.c:1265:40: warning: dereferencing type-punned pointer might break strict-aliasing rules [-Wstrict-aliasing]
 1265 | #define smallbin_at(M, i)   ((sbinptr)((char*)&((M)->smallbins[(i)<<1])))
      |                                       ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/src/dos/dosemu2/build/../src/base/lib/misc/dlmalloc.c:2268:19: note: in expansion of macro ‘smallbin_at’
 2268 |     sbinptr bin = smallbin_at(m,i);
      |                   ^~~~~~~~~~~
/usr/local/src/dos/dosemu2/build/../src/base/lib/misc/dlmalloc.c: In function ‘prepend_alloc’:
/usr/local/src/dos/dosemu2/build/../src/base/lib/misc/dlmalloc.c:1265:40: warning: dereferencing type-punned pointer might break strict-aliasing rules [-Wstrict-aliasing]
 1265 | #define smallbin_at(M, i)   ((sbinptr)((char*)&((M)->smallbins[(i)<<1])))
      |                                       ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/src/dos/dosemu2/build/../src/base/lib/misc/dlmalloc.c:1427:38: note: in definition of macro ‘RTCHECK’
 1427 | #define RTCHECK(e)  __builtin_expect(e, 1)
      |                                      ^
/usr/local/src/dos/dosemu2/build/../src/base/lib/misc/dlmalloc.c:1962:26: note: in expansion of macro ‘smallbin_at’
 1962 |   else if (RTCHECK((F == smallbin_at(M,I) || ok_address(M, F)) &&\
      |                          ^~~~~~~~~~~
/usr/local/src/dos/dosemu2/build/../src/base/lib/misc/dlmalloc.c:2151:20: note: in expansion of macro ‘unlink_small_chunk’
 2151 |   if (is_small(S)) unlink_small_chunk(M, P, S)\
## and many more, ending
/usr/local/src/dos/dosemu2/build/../src/base/lib/misc/dlmalloc.c:3309:11: note: in expansion of macro ‘insert_chunk’
 3309 |           insert_chunk(fm, p, psize);
      |           ^~~~~~~~~~~~
0x7f9f27b4ad09 __libc_start_main
	../csu/libc-start.c:308
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <file:///usr/share/doc/gcc-10/README.Bugs> for instructions.
```

--- a/Makefile
+++ b/Makefile
@@ -62,7 +62,7 @@ rpm: dosemu2.spec.rpkg
 	rpkg local
 
 deb:
-	debuild -i -us -uc -b
+	debuild -e CC=clang -i -us -uc -b
 
 changelog:
 	if [ -d $(top_srcdir)/.git -o -f $(top_srcdir)/.git ]; then \

